LearningMissionSimulation/PlaygroundEdvard.cs

```diff
@@ -252,7 +252,45 @@ public void AssignInstructorsToClassrooms()
 
         public void RegisterStudentsForClasses()
         {
-            throw new NotImplementedException();
+            if (ClassroomDictionary.Count == 0)
+            {
+                ReportError(missingResource: "Classroom", failedAction: "Register");
+            }
+            else if (ActiveStudentList.Count == 0)
+            {
+                ReportError(missingResource: "Student", failedAction: "Register");
+            }
+            else
+            {
+                int maxCount = 0;
+                foreach (Classroom classroom in ClassroomDictionary.Values)
+                {
+                    if (classroom.ItemList.Count < classroom.MaximumCapacity)
+                    {
+                        UpdateStudentList(classroom);
+                        maxCount = classroom.ItemList.Count;
+                    }
+                }
+            }
+        }
+
+        void UpdateStudentList(Classroom classroom)
+        {
+            int studentCount = AttributeGenerator.random.Next(0, (classroom.MaximumCapacity - classroom.ItemList.Count - ActiveStudentList.Count) + 1);
+            int count = 0;
+            while (count < studentCount)
+            {
+                foreach (Student student in StudentDictionary.Values)
+                {
+                    if (!student.ClassroomList.Contains(classroom))
+                    {
+                        student.ClassroomList.Add(classroom);
+                        classroom.ItemList.Add(student);
+                    }
+
+                    count++;
```

Comment:
You should check if the count exceeding studentCount.

Reply:
Done.